tracing/tracing.go

```diff
@@ -0,0 +1,121 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// propagator to use
+var propagator propagation.TextMapPropagator = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}) // traceparent header
+type Option interface {
+	apply(*options)
+}
+type options struct {
+	address string
+}
+
+// WithAddress takes an address in the form of Host:Port
+func WithAddress(addr string) Option {
+	return addrOption{address: addr}
+}
+
+type addrOption struct {
```

Comment:
Not a required change for this PR: gRPC demonstrates a good pattern for generating options without needing to define a struct for each one – https://github.com/grpc/grpc-go/blob/master/dialoptions.go#L105

Reply:
Oooh, a little obfuscated...but then again so is making a ton of structs and this is a lot easier to maintain. Will keep in mind for the future.